conversational-client/src/app/app.component.ts

```diff
@@ -15,34 +17,73 @@ export class AppComponent {
   speechRecognizerComponent!: SpeechRecognizerComponent;
   @ViewChild(ModeSelectorComponent)
   modeSelectorComponent!: ModeSelectorComponent;
-  @ViewChild('content', {read: ElementRef}) contentElement!: ElementRef;
+  @ViewChild('content', {read: ElementRef})
+  contentElement!: ElementRef;
+  @ViewChild(HistoryTutorComponent)
+  historyTutorComponent!: HistoryTutorComponent;
+  @ViewChild(FakeModeComponent)
+  fakeModeComponent!: FakeModeComponent;
 
   conversation: ChatMessage[] = [];
   interimDialogLine: string = '';
+  worldState: object = {};
```

Comment:
My thinking is that we may want to manipulate it on the client as well as the server (but I don't have a concrete need for that yet)

It's controlled / defined by the mode on the server -- it's generated by the llm based on a prompt... I didn't want to use a string since at the very least we need to parse / process it on the client, possibly change it.

It's really a dict... but whether the values are arrays / strings / objects I think depends on the mode.

I don't think any modes will share state... so it's an object model shared between the server & client side & probably different for each mode.

I'll add a TODO for now but agree we should come back to it.

Reply:
sounds good. I'll think of solutions to improve type safety, since relying on the model for type safety is fragile (ie. no guarantees that the string will parse as JSON).